### DISCUSSION TITLE
cert file problem

### DISCUSSION THREAD

**Discussion by linchan6369** [XTLS/Xray-core#1462] [Q&A]
hi
i instal xui and i can see cert and key file at root
but when i put this directory on xui pannel and try to save see this error everytime

Failed to modify settings: cert file </root/cert.crt > or key file </root/private.key> invalid: open /root/cert.crt : no such file or directory

**Comment by cross-hello** [comment]
So maybe you could modify user of cert and key file via `chown user_please_specify_your_xui_owner /root/*`.
Or extend file read right via `chmod o+r /root/*`


Dec 26, 2022 23:21:52 linchan6369 ***@***.***>:

**Reply by Alireza-Samadi** [reply]
I have same problem and your solution have not helped me

**Reply by cross-hello** [reply]
😏 just share some suggestions, why must need to be helpful

**Reply by a-nahvi** [reply]
Hi, I have the same problem.
you said modify user of cert and key or extend fire read light
how can I do that?
I mean can you explain step by step.
thanks

**Reply by cross-hello** [reply]
```sh
#modify user of the cert and key file
chown user_please_specify_your_xui_owner /root/*
#extend file read right
chmod o+r /root/*
```

Sorry, Sir. May it not be detailed as possible, we want back to exercise violin.

**Comment by chika0801** [comment]
参考
https://github.com/v2fly/fhs-install-v2ray/wiki/Insufficient-permissions-when-using-certificates
